examples/paho-mqtt/Makefile

```
@@ -33,25 +33,20 @@ USEMODULE += ps
 USEMODULE += netdev_default
 USEPKG += paho-mqtt
 
-# paho-mqtt depends on TCP support, choose which stacks you want
-LWIP_IPV4 ?= 0
-LWIP_IPV6 ?= 1
-
-ifneq (0,$(LWIP_IPV4))
-  USEMODULE += ipv4_addr
-  USEMODULE += lwip_arp
-  USEMODULE += lwip_ipv4
-  USEMODULE += lwip_dhcp_auto
-  CFLAGS += -DETHARP_SUPPORT_STATIC_ENTRIES=1
-endif
-
-ifneq (0,$(LWIP_IPV6))
-  USEMODULE += ipv6_addr
-  USEMODULE += lwip_ipv6_autoconfig
-endif
-
-USEMODULE += lwip_netdev
-USEMODULE += lwip
+USEMODULE += auto_init_gnrc_netif
+# Activate ICMPv6 error messages
+USEMODULE += gnrc_icmpv6_error
+# Specify the mandatory networking module for a IPv6 routing node
+USEMODULE += gnrc_ipv6_router_default
```

```suggestion
USEMODULE += gnrc_ipv6_default
```

should be enough here

---

I actually thought about which GNRC modules are required and which are optional and came to the conclusion that for the ease-of-use requirement adding routing support is beneficial. Same goes for ICMP. If someone wants to the test the application in the IoT-Lab, for instance, having RPL and routing active might be helpful to actually get a route towards the 6lbr - and having ICMP enabled is beneficial to check the connectivity first.

Of course, we could also remove the modules and try to keep the example application as slim as possible.

Both options have their pros and cons IMO.

---

The thing is that this will actually make it *harder* to use. When the node is a router, it will not send router solicitations. That means you’ll have to wait for the next unsolicited router advertisement to get a global address.

---

Valid point. So, how about keeping RPL, but switching to host, i.e., RPL leaf node?

---

tbh I'm not sure if RPL actually is used much at all, at appears to be rather abandoned.

---

Are you talking about usage or maintenance? As far as I can tell, it is working well.

---

If @benpicco isn't using it, it is abandoned ;) I use it regularly in my setups.

---

(and it is used with gnrc_networking, so most people using that example will use it)

---

@miri64 are you only using it with static topologies? It appears that repairing the DODAG  when a node vanishes just doesn't happen, which is kinda crucial in real-world szenarios.
When I asked about it more than a year ago I couldn't find anyone who used it outside simple / short lived test scenarios and nobody who maintains the module.

---

If this bug is still present basically the same arguments from below (whether to enable LWIP as an alternative or not) apply. If RPL works in some scenarios it is nice to have for this application, but it may get frustrating for users using it an dynamically changing topology. I used RPL on real hardware with bigger topologies only in the IoT-LAB so far, so I've observed the bug myself.